src/main.rs

```diff
@@ -382,22 +374,48 @@ fn external_bindle_id(entry: &HippoFactsEntry) -> Option<bindle::Id> {
     entry.external_ref().map(|ext| ext.bindle_id)
 }
 
-/// Find the source directory
-fn sourcedir(hippofacts_arg: &str) -> anyhow::Result<PathBuf> {
-    let source_file_or_dir = std::env::current_dir()?.join(hippofacts_arg);
-    let source = if source_file_or_dir.is_file() {
-        source_file_or_dir
-    } else {
-        source_file_or_dir.join("HIPPOFACTS")
-    };
+fn hippofacts_file_path_from_args(args: &ArgMatches) -> anyhow::Result<PathBuf> {
+    args.value_of(ARG_HIPPOFACTS)
+        .ok_or_else(|| anyhow::Error::msg("HIPPOFACTS file is required"))
+        .and_then(hippofacts_file_path)
+}
 
-    if !source.exists() {
-        return Err(anyhow::anyhow!(
+fn hippofacts_file_path(hippofacts_arg: &str) -> anyhow::Result<PathBuf> {
+    let source = std::env::current_dir()?.join(hippofacts_arg);
+    if source.is_dir() {
+        find_hippofacts_file_in(&source)
+    } else if source.is_file() {
+        Ok(source)
+    } else {
+        Err(anyhow::anyhow!(
             "Artifacts spec not found: file {} does not exist",
             source.to_string_lossy()
-        ));
+        ))
+    }
+}
+
+const SPEC_FILENAMES: &[&str] = &[
```

Comment:
do we want to accept PascalCase, snake_case and kebab-case style filenames?

```
HippoFacts
HippoFacts.toml
hippo_facts
hippo_facts.toml
hippo-facts
hippo-facts.toml
```

Reply:
I wondered about `HippoFacts` or just trying any combination of cases a la `hIpPoFaCtS`.  The latter felt like too much work for too little return.  The former looked, I dunno, wronger than I expected.  But I can do it if you want.

I don't think we need snake or kebab case.  If the user _really_ wants that then they can always `hippo push ./hippo-facts.toml` explicitly.